joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -1530,6 +1532,26 @@ bool JointTrajectoryController::has_active_trajectory() const
   return traj_external_point_ptr_ != nullptr && traj_external_point_ptr_->has_trajectory_msg();
 }
 
+void JointTrajectoryController::update_pids()
+{
+  for (size_t i = 0; i < dof_; ++i)
+  {
+    const auto & gains = params_.gains.joints_map.at(params_.joints[i]);
+    if (pids_.at(i))
```

Comment:
```suggestion
    if (pids_[i])
```
Why not use `pids_[i]` directly here?

Reply:
This was an intermediate step while I was looking for the cause of a segfault ;) Is it bad practice to use .at() method of `std::vector`?

Reply:
Well it is not a bad practice, but accessing by .at() is a bit slower than the other way.

Reply:
Anyway you are already resizing your vector at the configure phase, I think it shouldn't cause seg fault.

Reply:
It did without the check if  `use_closed_loop_pid_adapter_==true`, because the PID vector is empty then. As I said, I was looking for the source of the segfault, now it should be save to skip.